lms/djangoapps/course_api/blocks/serializers.py

```diff
@@ -67,6 +72,20 @@ def to_representation(self, block_key):
                     supported_field.default_value,
                 )
                 if field_value is not None:
+
+                    # override due date with student extension in unit added by instructor.
+                    if supported_field.block_field_name is 'due':
+                        override = StudentFieldOverride.objects.filter(
+                            course_id=block_key.course_key,
+                            student=self.context['request'].user,
+                            location=block_key,
+                            field='due'
+                        ).first()
+
```

Comment:
I just want to confirm: Is this how the date overrides are implemented elsewhere?

Reply:
can you give reference where it is on other side in edx ?

Reply:
I believe,  we need to look around the `StudentFieldOverride` model to get the extension date.

Reply:
I'm not sure because it was changed since we last worked on it. But the due date display works properly when the learner views the sequence. 

You might look at this: https://github.com/edx/edx-platform/pull/19848 and https://github.com/edx/edx-platform/pull/20121 -- I think they are include in Ironwood, but you should check to be sure.

Reply:
logically, It seems me the query above always return a single record for a particular student, unit and course for `due` field. If you think then I can request someone else for MIT to review and test the change set ?

Reply:
I'm not really concerned about the logic. I'm more concerned about the code pattern. I would like to submit this upstream to edX and they are more likely to accept if the approach is the same as used elsewhere. Among other things, they are very sensitive about performance and if you do it the same way then it probably doesn't degrade performance. 

Also, you'll need tests.